staging/src/k8s.io/apiserver/pkg/server/options/authorization.go

```diff
@@ -84,31 +95,43 @@ func (s *DelegatingAuthorizationOptions) ApplyTo(c *server.AuthorizationInfo) er
 		return nil
 	}
 
-	cfg, err := s.ToAuthorizationConfig()
-	if err != nil {
-		return err
-	}
-	authorizer, err := cfg.New()
+	authzr, err := s.ToAuthorization()
 	if err != nil {
 		return err
 	}
-
-	c.Authorizer = authorizer
+	c.Authorizer = authzr
 	return nil
 }
 
-func (s *DelegatingAuthorizationOptions) ToAuthorizationConfig() (authorizerfactory.DelegatingAuthorizerConfig, error) {
+func (s *DelegatingAuthorizationOptions) ToAuthorization() (authorizer.Authorizer, error) {
+	var authorizers []authorizer.Authorizer
+
+	if len(s.AlwaysAllowPaths) > 0 {
+		a, err := path.NewAuthorizer(s.AlwaysAllowPaths)
+		if err != nil {
+			return nil, err
+		}
+		authorizers = append(authorizers, a)
+	}
+
 	sarClient, err := s.newSubjectAccessReview()
 	if err != nil {
-		return authorizerfactory.DelegatingAuthorizerConfig{}, err
+		return nil, err
 	}
-
-	ret := authorizerfactory.DelegatingAuthorizerConfig{
-		SubjectAccessReviewClient: sarClient,
-		AllowCacheTTL:             s.AllowCacheTTL,
-		DenyCacheTTL:              s.DenyCacheTTL,
+	if sarClient != nil {
```

Comment:
`sarClient` should be non-nil if `err == nil`, remove this check

Reply:
true. In a follow-up PR it might be nil. Fixed here.